### PR TITLE
Add splitPath operation

### PR DESCRIPTION
**Pull request by harendra-kumar**

### CODE REVIEW

File: core/src/Streamly/Internal/FileSystem/Path/Common.hs [hunk 1]
```diff
@@ -463,3 +484,42 @@ append :: (Unbox a, Integral a) =>
...
+-- components in the path are ignored except in the leading position. Multiple
+-- consecutive separators are ignored.
+--
+-- >>> :{
+--  splitPath Common.Posix = Stream.toList . fmap unpackPosix . Common.splitPath Common.Posix . packPosix
+--  splitPath Common.Windows = Stream.toList . fmap unpackWindows . Common.splitPath Common.Windows . packWindows
+-- :}
+--
+-- >>> splitPath Common.Posix "."
+-- ["."]
```
Comment (adithyaov) [inline core/src/Streamly/Internal/FileSystem/Path/Common.hs:498]: Do the doctests pass.
How do we keep the leading `.`?
We are filtering them out right?

Perhaps `off /= 0` should be a part of `shouldFilterOut`
```
    shouldFilterOut (off, len) =
        len == 0 ||
            (len == 1 && unsafeIndexChar off arr == '.' && off /= 0)
```

File: core/src/Streamly/Internal/FileSystem/Path/Common.hs [hunk 2]
```diff
@@ -463,3 +538,284 @@ append :: (Unbox a, Integral a) =>
...
+-- ["/","home"]
+--
+-- >>> splitPosix "home//user/./..////\\directory/."
+-- ["home","user","..","\\directory"]
+--
+-- >>> splitWin "home//user/./..////\\directory/."
+-- ["home","user","..","directory"]
+--
+{-# INLINE splitPath #-}
+splitPath
```
Comment (adithyaov) [inline core/src/Streamly/Internal/FileSystem/Path/Common.hs:785]: `splitPath` will ignore the trailing separator.
Is there a loss of information here?

Comment (adithyaov) [reply core/src/Streamly/Internal/FileSystem/Path/Common.hs:785]: `a/b/` is the same as `a/b/.`
`a/b` is different from `a/b/`

Comment (harendra-kumar) [reply core/src/Streamly/Internal/FileSystem/Path/Common.hs:785]: Ideally the components should not have separators. We can keep the trailing slash but that would require special casing it and not sure how useful that will be.

**Pull request merged**